rusoto/core/src/client.rs

```diff
@@ -148,115 +141,52 @@ impl<P, D> Clone for ClientInner<P, D> {
     }
 }
 
-impl<P, D> SignAndDispatch for ClientInner<P, D>
+async fn sign_and_dispatch<P, D>(
+    client: ClientInner<P, D>,
+    request: SignedRequest,
+    timeout: Option<Duration>,
+) -> Result<HttpResponse, SignAndDispatchError>
 where
     P: ProvideAwsCredentials + Send + Sync + 'static,
-    P::Future: Send,
     D: DispatchSignedRequest + Send + Sync + 'static,
-    D::Future: Send,
 {
-    fn sign_and_dispatch(
-        &self,
-        request: SignedRequest,
-    ) -> Box<dyn TimeoutFuture<Item = HttpResponse, Error = SignAndDispatchError> + Send> {
-        Box::new(SignAndDispatchFuture {
-            inner: self.clone(),
-            state: Some(SignAndDispatchState::Lazy { request }),
-            timeout: None,
-        })
-    }
-}
-
-pub struct SignAndDispatchFuture<P: ProvideAwsCredentials, D: DispatchSignedRequest> {
-    inner: ClientInner<P, D>,
-    state: Option<SignAndDispatchState<P, D>>,
-    timeout: Option<Duration>,
+    let p = client.credentials_provider.clone().unwrap();
```

Comment:
(This particular `unwrap` seems to be the culprit of the `without_signature` test failure.)